### ISSUE TITLE
Blender (for both 32-bit & 64-bit)

### ISSUE THREAD

**Issue by fabianmendes** [Botspot/pi-apps#108]
Hi. [App request]

Botspot, I've made the scrip but I think you may take a look forward into it, because I'm not sure if everything is right, also, it seems to be better on changing the Icon. Here is the zip file:
[blender_2.79b.zip](https://github.com/Botspot/pi-apps/files/5606088/blender_2.79b.zip)

**Comment by Botspot** [comment]
Isn't Blender already installable via `apt-get`?

Pi-Apps currently only allows software that's on the `apt` repositories.

**Comment by Itai-Nelken** [comment]
The newest version of blender is 2.91.0, and the version on `apt` and the version this script installs are 2.79, it would be nice to have a newer version on pi-apps because some programs that use blender need the newer version, but I'm not sure it's necesary.

**Comment by fabianmendes** [comment]
No, The newest versions doesnt work on raspberry pi, I've tried all them. The only one versión that works properly is this one. Thats why

**Comment by Botspot** [comment]
Mmm, ok. Yes, I do allow newer versions of stuff to be added to Pi-Apps, as long as the newer version is actually something a majority of users would prefer.

I'll take a look.

**Comment by Itai-Nelken** [comment]
I'm trying to build the newer version of blender on the pi. I'm pretty sure it won't work with the instructions they provide, but still worth trying.

**Comment by Botspot** [comment]
@fabianmendes 
In the uninstall script, you should not add anything to the purge-installed command.
```
"${DIRECTORY}/purge-installed" "blender-data" "$(dirname "$0")" || exit 1
```
It doesn't work that way.  
Purge-installed remembers what Blender installed with `pkg-install`, and automatically removes those packages.

**Comment by Itai-Nelken** [comment]
> I'm trying to build the newer version of blender on the pi. I'm pretty sure it won't work with the instructions they provide, but still worth trying.

Doesn't work

**Comment by Itai-Nelken** [comment]
> No, The newest versions doesnt work on raspberry pi, I've tried all them. The only one versión that works properly is this one. Thats why

The version your scripts installs is the same available from `apt`

**Comment by Botspot** [comment]
Yeah APT can already install 2.79 from the repos. So adding a 2.79 version to Pi-Apps is pointless.

@fabianmendes If you find a way to get a later version of Blender going, then Pi-Apps could host it.

**Comment by fabianmendes** [comment]
> Yeah APT can already install 2.79 from the repos. So adding a 2.79 version to Pi-Apps is pointless.
> 
> @fabianmendes If you find a way to get a later version of Blender going, then Pi-Apps could host it.

Cool, right. I didn't know. Ok, all right!

**Comment by ThaPg** [comment]
When i tried to install Blender using command line `sudo apt install blender` the downloaded version was 2.83.5 . This version is not working on Raspberry PI 4 with 4GB RAM. As @Botspot  mentioned in https://github.com/Botspot/pi-apps/issues/288, the latest available blender version for RPi4 is 2.79b due to OpenGL nonsupport for the processor. 

I downloaded the .zip file from @fabianmendes but the script is not working. After unzipped the file, the execution is closing and can't continue.

If someone try to find the solution, maybe will be good for the community members with Raspberry Pi 4.

**Comment by Botspot** [comment]
@ThaPg, you're right! Looks like the upgrade to Bullseye broke Blender. 

Once we work out how to get it running, I'd expect this to be added to Pi-Apps.
For now, see: https://forums.raspberrypi.com/viewtopic.php?p=1942999

**Comment by theofficialgman** [comment]
has anyone tried to override the mesa reported opengl version and use blender 2.83?
the rpi v3d opengl drivers support a lot of the featureset of opengl up to 4.3, its possible that the right extensions are implemented and blender could run
```
MESA_GL_VERSION_OVERRIDE=4.3 blender
```

**Comment by Botspot** [comment]
> has anyone tried to override the mesa reported opengl version and use blender 2.83? the rpi v3d opengl drivers support a lot of the featureset of opengl up to 4.3, its possible that the right extensions are implemented and blender could run
> 
> ```
> MESA_GL_VERSION_OVERRIDE=4.3 blender
> ```

I just uninstalled blender and closed the terminal, but from what I remembered, blender said it needed OpenGL 3.3. So I set the override variable to 3.3. Blender briefly launched, but then crashed. The terminal simply said "Bus error".

**Comment by theofficialgman** [comment]
the blender website (right now) says 4.3 is required, I'm not aware which exact extentions they are using but it might be interesting to still check. I somehow doubt the bus error is coming from something lacking from opengl though...

either way, blender 3.X WILL be using vulkan and metal only, no opengl, by sometime in later 2022
so this issue is only temporary until that version is released

**Comment by cycool29** [comment]
> ```
> MESA_GL_VERSION_OVERRIDE=4.3 blender
> ```

This works for me.

![image](https://user-images.githubusercontent.com/88134003/148006020-8588a58b-5f17-4aa8-8b0d-4038011a5295.png)

**Comment by theofficialgman** [comment]
@cycool29 I'm not a blender user, but could you trying doing a few things in the software and see if anything crashes (due to missing opengl features)

**Comment by Botspot** [comment]
> > ```
> > MESA_GL_VERSION_OVERRIDE=4.3 blender
> > ```
> 
> This works for me.

4.3 doesn't work for me. I wonder what the difference is.
```
pi@raspberrypi:~ $ MESA_GL_VERSION_OVERRIDE=4.3 blender
Warning: Could not find a matching GPU name. Things may not behave as expected.
Detected OpenGL configuration:
Vendor: Broadcom
Renderer: V3D 4.2
Bus error
```

**Comment by cycool29** [comment]
> @cycool29 I'm not a blender user

Nor do I.

**Comment by cycool29** [comment]
> 4.3 doesn't work for me. I wonder what the difference is.
> 
> ```
> pi@raspberrypi:~ $ MESA_GL_VERSION_OVERRIDE=4.3 blender
> Warning: Could not find a matching GPU name. Things may not behave as expected.
> Detected OpenGL configuration:
> Vendor: Broadcom
> Renderer: V3D 4.2
> Bus error
> ```

Weird. Even `MESA_GL_VERSION_OVERRIDE=3.2 blender` works for me.

**Comment by Botspot** [comment]
> Weird. Even `MESA_GL_VERSION_OVERRIDE=3.3 blender` works for me.

What's the full output on your system?

**Comment by cycool29** [comment]
> What's the full output on your system?

```
pi@raspberrypi:~ MESA_GL_VERSION_OVERRIDE=3.3 blender
Warning: Could not find a matching GPU name. Things may not behave as expected.
Detected OpenGL configuration:
Vendor: Broadcom
Renderer: V3D 4.2
```

**Comment by Botspot** [comment]
I guess if someone wants to make this a pi-app, we can try to automate this tutorial for installing Buster-Blender on bullseye. https://forums.raspberrypi.com/viewtopic.php?p=1942999

**Comment by cycool29** [comment]
> 

Why not just install it from APT? It has newer version.

**Comment by theofficialgman** [comment]
> I guess if someone wants to make this a pi-app, we can try to automate this tutorial for installing Buster-Blender on bullseye. https://forums.raspberrypi.com/viewtopic.php?p=1942999

I very much would not approve of the process in that thread for the same reasons mentioned there
it would be better to have an appimage/flatpak/snap of blender 2.79 with all of those libraries contained rather than doing them systemwide and risking breaking other stuff

**Comment by ThaPg** [comment]
Thank you very much for your replies.
Could someone tell me the command to see my MESA_GL_VERSION_OVERRIDE. It's from command `glxinfo`?
Later, maybe I'll install again the rasbian and i'll try to install vulkan drivers (in a "clean" machine) and then the blender again using tutorials (https://qengineering.eu/install-vulkan-on-raspberry-pi.html and https://www.youtube.com/watch?v=j5mh-joIFJ4&ab_channel=JoseCerrejon). (sorry if it's forbidden to add links)

@theofficialgman thanks for reply about future drivers for blender and vulkan.

**Comment by ThaPg** [comment]
Hallo again. I installed again "blender". Then I copied `MESA_GL_VERSION_OVERRIDE=4.3 blender` and blender was started. But it doesn't begin from icon. It begins only if i write the previous command in terminal. A little bit strange!!

**Comment by Botspot** [comment]
> Hallo again. I installed again "blender". Then I copied `MESA_GL_VERSION_OVERRIDE=4.3 blender` and blender was started. But it doesn't begin from icon. It begins only if i write the previous command in terminal. A little bit strange!!

Yes, running from the menu won't apply that override. You can edit the menu launcher though.

**Comment by ThaPg** [comment]
I update my answer with terminal print. Thanks for the fast response.

**Comment by ThaPg** [comment]
I don't know if it is possible, but someone make a script following next steps(i can write script the first two steps):
1) Download and install blender
2) Download and install mesa-utils
3) Make blender starts using `MESA_GL_VERSION_OVERRIDE=4.3 blender`

That's my thoughts. I don't know if you agree or disagree with my thoughts. I don't know also if mesa-utils will make problem with other programms.

Also, I downloaded PiKiss, I installed Vulkan and I didn't see any changes in `glxinfo`, maybe because I have already installed mesa-utils.

**Comment by theofficialgman** [comment]
@ThaPg glxinfo will NEVER show any vulkan info, its an opengl info viewer, thats it

could you please test out some of the blender functionality to make sure it doesn't crash on the PI? I'm not confident using `MESA_GL_VERSION_OVERRIDE=4.3 blender` will actually make all the software work.

**Comment by ThaPg** [comment]
Ooops yeah @theofficialgman . My fault. I'm sorry.

I'll try to run blender for some days and in different  situations. I'll give feedback here.

**Comment by fabianmendes** [comment]
Mhm, I saw the error: (I've "installed" gdebi, but written as comment)
![2022-01-13-102025_800x480_scrot](https://user-images.githubusercontent.com/31230743/149347248-625bcc46-a383-4b2a-9eec-dd2ce3b684b9.png)

Here's again the script. And using pkg-install
[Blender.zip](https://github.com/Botspot/pi-apps/files/7863539/Blender.zip)
Please try this. It's only for 32-bit, also It doesn't have a description nor how to run instructions as a pi-app introduction. And I didn't make its Desktop entry.

**Comment by theofficialgman** [comment]
> Here's again the script. And using pkg-install 
@fabianmendes pkg-install is legacy support only as of over 2 months ago, do not use it, the pi-apps documentation needs updating
install_packages is the new format, look at any other pi-apps script for reference for how to use it

**Comment by fabianmendes** [comment]
> > Here's again the script. And using pkg-install
> 
> @fabianmendes pkg-install is legacy support only as of over 2 months ago, do not use it, the pi-apps documentation needs updating install_packages is the new format, look at any other pi-apps script for reference for how to use it


oh alright! Thaanks

Otherwise, did the script work?

**Comment by cycool29** [comment]
> Otherwise, did the script work?


No, the binary is in `i686` and box86 shows these error when launching:
```
Box86 with Dynarec v0.2.5 b903e57e built on Jan 16 2022 00:07:22
Error: Global Symbol __RML_open_factory not found, cannot apply R_386_GLOB_DAT @0xfd4a690 ((nil)) in ./blender
Error: Global Symbol __TBB_make_rml_server not found, cannot apply R_386_GLOB_DAT @0xfd4a694 ((nil)) in ./blender
Error: Global Symbol __RML_close_factory not found, cannot apply R_386_GLOB_DAT @0xfd4a698 ((nil)) in ./blender
Error: Global Symbol __TBB_call_with_my_server_info not found, cannot apply R_386_GLOB_DAT @0xfd4a69c ((nil)) in ./blender
```

**Comment by fabianmendes** [comment]
Why is Box86 making excuses? I mean, I don't understand that log.

**Comment by theofficialgman** [comment]
also @fabianmendes , I don't think anyone really wants to use blender through box86, its an open source package and it can be made to build/install on armhf/arm64
I responded to you initially with just reading your script and didn't realize you had commented on the blender issue topic, I would have suggested you stop developing your script if it had registered that this was for blender

**Comment by theofficialgman** [comment]
@Botspot please try blender again with this and tell the output:
```
MESA_GL_VERSION_OVERRIDE=4.3 blender --debug-all
```

**Comment by fabianmendes** [comment]
> also @fabianmendes , I don't think anyone really wants to use blender through box86, its an open source package and it can be made to build/install on armhf/arm64 I responded to you initially with just reading your script and didn't realize you had commented on the blender issue topic, I would have suggested you stop developing your script if it had registered that this was for blender

I didn't get this comment/reply.
And you're right, I don't know why Box86 came out... I don't use Box86 for that, I just had surprised me! There is no reason to use Box86 to run Blender at all!!

**Comment by Botspot** [comment]
> @Botspot please try blender again with this and tell the output:
> 
> ```
> MESA_GL_VERSION_OVERRIDE=4.3 blender --debug-all
> ```

Great. Now it worked. I think it's working now because I'm temporarily using the armhf kernel, rather than the 64-bit kernel that I usually use. Can someone verify this?

**Comment by derryktutt** [comment]
Blender does not work with MESA_GL_VERSION_OVERRIDE=4.3 blender --debug-all. When I click on View port Shading, the app just freezes.

**Comment by ThaPg** [comment]
Sorry for being late to my answer. I tried to `MESA_GL_VERSION_OVERRIDE=4.3 blender` sometimes. Unfortunately, as @theofficialgman told, I had problems to run some blender applications. Also I have the same problem when i use Cura (3d slicer). All problems are coming from graphic issues.

**Comment by theofficialgman** [comment]
I just wanted to add, this is the issue tracking vulkan support in blender: https://developer.blender.org/T68990

it may not work on the raspberry pi though, I am not sure if the pi support the necessary vulkan extensions they plan to use or not

**Comment by virat2010** [comment]
well, that's sad. I love blender! I use it on my other computer any way!

**Comment by MeltingDiamond** [comment]
After installing blender using official install script I can run it using: LIBGL_ALWAYS_SOFTWARE=true blender
